### DISCUSSION TITLE
Published Ports are not accessible via IP address

### DISCUSSION THREAD

**Discussion by engageub** [xjasonlyu/tun2socks#368] [Q&A]
Hello,
When I try to access published ports via IP:PORT in tun2socks, it is not accessible where as using 127.0.0.1:PORT and 0.0.0.0:PORT are working fine. net.ipv4.ip_forward is also set 1 in the container.

If I use published ports on direct docker image instead of tun2socks, it is accessible via IP:PORT. So, the only issue is accessing via IP:PORT on tun2socks container. 

To Test this, you can use [InternetIncome test branch](https://github.com/engageub/InternetIncome/tree/test).
Set the following in properties.conf file for directly accessing the container without tun2socks
```
 Set CUSTOM_CHROME to true 
 Set USE_DIRECT_CONNECTION to true
```
After starting the container, you can try accessing via IP:PORT. It works fine.

Now, to access via tun2socks proxy. 
Set the following in properties.conf

```
 Set CUSTOM_CHROME to true 
 Set USE_DIRECT_CONNECTION to false
 Set USE_PROXIES to true
```
Set a proxy inside  proxies.txt file. 
After starting the container, you will notice that it is accessible via 127.0.0.1:IP and 0.0.0.0:IP but not IP:PORT of VPS.

Could you please suggest what configuration has to be enabled to access via IP:PORT through tun2socks container. 

Thank you

**Comment by engageub** [comment]
Hi,
To present this in simple terms, the following is accessible via IP:PORT where as when used with tun2socks its not accessible.

```
docker run -d --name alpi -p 7000:3200 alpine sh -c "while true; do sleep 3600; done"
sudo docker run -d --name chrome --network=container:alpi --security-opt seccomp=unconfined -e TZ=Etc/UTC -e CUSTOM_HTTPS_PORT=3201 -e CUSTOM_PORT=3200 --shm-size="1gb" lscr.io/linuxserver/chromium:latest
```
The following uses tun2socks and is not accessible via IP:PORT although it is accessible via 127.0.0.1:PORT and 0.0.0.0:PORT

```
sudo docker run -d --name tun2socks -e PROXY=http://2.3.4.5:8000 -v '/dev/net/tun:/dev/net/tun' --cap-add=NET_ADMIN -p 7000:3200 xjasonlyu/tun2socks:v2.5.2
sudo docker run -d --name chrome --network=container:tun2socks --security-opt seccomp=unconfined -e TZ=Etc/UTC -e CUSTOM_HTTPS_PORT=3201 -e CUSTOM_PORT=3200 --shm-size="1gb" lscr.io/linuxserver/chromium:latest
```

**Reply by xjasonlyu** [reply]
Hmmm, interesting.. Not sure why 🤔

**Reply by engageub** [reply]
Thank you for the response. I raised this issue in [docker forums](https://forums.docker.com/t/published-ports-on-tun2socks-container-are-not-accessible-via-host-ip-address/142201) and they redirected to contact tun2socks. However, this is happening with tun2proxy as well. 
So some network configuration might be blocking something.

**Reply by inkpool** [reply]
Did you resolve this issue? I met the same issue.

**Reply by engageub** [reply]
Hello,
I am using https://serveo.net/ to access the server externally when required, as localhost(127.0.0.1)  is accessible. 
The issue is still not resolved. 

Thank you
